src/client/clientenvironment.cpp

```diff
@@ -500,14 +500,20 @@ void ClientEnvironment::getSelectedActiveObjects(
 		if (!obj->getSelectionBox(&selection_box))
 			continue;
 
-		const v3f &pos = obj->getPosition();
-		aabb3f offsetted_box(selection_box.MinEdge + pos,
-			selection_box.MaxEdge + pos);
-
 		v3f current_intersection;
-		v3s16 current_normal;
-		if (boxLineCollision(offsetted_box, shootline_on_map.start, line_vector,
-				&current_intersection, &current_normal)) {
+		v3f current_normal;
+		v3f rel_pos = shootline_on_map.start - obj->getPosition();
+		bool collision;
+		GenericCAO* gcao = dynamic_cast<GenericCAO*>(obj);
```

Comment:
The alternative to this downcast is to move the properties to the CAO class (or perhaps to merge GenericCAO and CAO altogether?), but such refactors are out of scope for this PR.